collections/src/main/scala/strawman/collection/immutable/ImmutableArray.scala

```diff
@@ -125,8 +125,8 @@ sealed abstract class ImmutableArray[+A]
   * @define coll immutable array
   * @define Coll `ImmutableArray`
   */
-object ImmutableArray extends StrictOptimizedClassTagSeqFactory[ImmutableArray] { self =>
-  val untagged: SeqFactory[ImmutableArray] = new ClassTagSeqFactory.AnySeqDelegate(self)
+object ImmutableArray extends StrictOptimizedClassTagSeqFactory[ImmutableArray] {
+  val untagged: SeqFactory[ImmutableArray] = new ClassTagSeqFactory.AnySeqDelegate(this)
```

Comment:
What's the reason for these changes? Just removing some unnecessary code, or is there a change in Dotty that necessitates this?

Reply:
We recently disallowed named self in lampepfl/dotty#4118. We might reconsider it though

Reply:
But the fix refers to the wrong this! Btw we will revert #4118.